src/Http/Response.php

```diff
@@ -764,16 +528,10 @@ public function getMimeType(string $alias): array|string|false
     public function mapType(array|string $ctype): array|string|null
     {
         if (is_array($ctype)) {
-            return array_map($this->mapType(...), $ctype);
-        }
-
-        foreach ($this->_mimeTypes as $alias => $types) {
-            if (in_array($ctype, (array)$types, true)) {
-                return $alias;
-            }
+            return array_map('\Cake\Http\MimeType::getExtension', $ctype);
```

Comment:
We might be missing a test for this
```
I get
TypeError: Cake\Http\MimeType::getExtension(): Argument #1 ($mimeType) must be of type string, array given

/data/work/git/cakephp-shim/vendor/cakephp/cakephp/src/Http/MimeType.php:352
/data/work/git/cakephp-shim/vendor/cakephp/cakephp/src/Http/Response.php:531
```

Reply:
```php
$content = new ContentTypeNegotiation();
$accept = $content->parseAccept($request);
$accepts = $response->mapType($accept);		
```
and apparently a result for parseAccept as
```
[
  '1.0' => [
    (int) 0 => 'text/csv'
  ],
  '0.8' => [
    (int) 0 => 'application/json'
  ],
  '0.7' => [
    (int) 0 => 'application/xml'
  ]
]
```
worked before

and now breaks here as the array_map() is not handling this anymore.